### ISSUE TITLE
Strange -Wmisleading-indentation

### ISSUE THREAD

**Issue by edigiacomo** [ARPA-SIMC/arkimet#39]
In #38 I have fixed some misleading indentations but there are more that I don't understand, because they seem typos (a semicolon after an `if`), but `git blame` says that they're very old commit...

- https://github.com/ARPA-SIMC/arkimet/blob/master/arki/dataset/gridquery.cc#L392
```
if (luaL_newmetatable(L, "arki.gridquery"));
{
...
```
- https://github.com/ARPA-SIMC/arkimet/blob/master/arki/dataset/index/summarycache.cc#L119
```
if (sys::unlink_ifexists(summary_pathname(t.ye, t.mo)));
    deleted = True
```
- https://github.com/ARPA-SIMC/arkimet/blob/master/arki/summary.cc#L218
```
if (luaL_newmetatable(L, "arki.summary"));
{
```
- https://github.com/ARPA-SIMC/arkimet/blob/master/arki/metadata.cc#1019
```
if (luaL_newmetatable(L, "arki.metadata"));
{
```

**Comment by spanezz** [comment]
They are typos, and it's code that could never have worked. See http://www.c2.com/cgi/wiki?SchroedinBug

I'll examine them one by one tomorrow.

**Comment by spanezz** [comment]
If in the meantime you want to keep compiling, you can replace `if (FOO);` with `FOO;` to get equivalent code.

**Comment by edigiacomo** [comment]
Thanks, but until the fix of #37 the code cannot be compiled, so @brancomat and I patiently wait :)

**Comment by spanezz** [comment]
Closed in da34c5a9010997f5d2c0aafc0dd3c5e56bdef6b0 (with the wrong bug number in the commit message)
